src/OrchardCore.Modules/OrchardCore.OpenId/Services/OpenIdBackgroundTask.cs

```diff
@@ -27,87 +25,33 @@ public OpenIdBackgroundTask(ILogger<OpenIdBackgroundTask> logger)
-                                await manager.GetIdAsync(authorization, cancellationToken));
-                        }
-
-                        catch (Exception exception)
-                        {
-                            _logger.LogDebug(exception,
-                                "An error occurred while removing the authorization {AuthorizationId} from the database.",
-                                await manager.GetIdAsync(authorization, cancellationToken));
-                        }
-                    }
-
-                    cancellationToken.ThrowIfCancellationRequested();
-                }
-
-                while (!authorizations.IsDefaultOrEmpty);
+                await serviceProvider.GetRequiredService<OpenIdAuthorizationManager>().PruneInvalidAsync(cancellationToken);
             }
-
-            async Task DeleteTokensAsync()
+            catch (OperationCanceledException exception) when (exception.CancellationToken == cancellationToken)
             {
-                // Note: the token manager MUST be resolved from the scoped provider as it
-                // depends on scoped stores that should be disposed as soon as possible.
-                var manager = serviceProvider.GetRequiredService<OpenIdTokenManager>();
-
-                ImmutableArray<IOpenIdToken> tokens;
-
-                do
-                {
-                    // Note: don't use an offset here, as the elements returned by this method
-                    // are progressively removed from the database immediately after calling it.
-                    tokens = await manager.ListInvalidAsync(100, 0, cancellationToken);
-
-                    foreach (var token in tokens)
-                    {
-                        cancellationToken.ThrowIfCancellationRequested();
-
-                        try
-                        {
-                            await manager.DeleteAsync(token, cancellationToken);
-
-                            _logger.LogDebug("The token {TokenId} was successfully removed from the database.",
-                                await manager.GetIdAsync(token, cancellationToken));
-                        }
-
-                        catch (Exception exception)
-                        {
-                            _logger.LogDebug(exception,
-                                "An error occurred while removing the token {TokenId} from the database.",
-                                await manager.GetIdAsync(token, cancellationToken));
-                        }
-                    }
-
-                    cancellationToken.ThrowIfCancellationRequested();
-                }
+                _logger.LogDebug("The authorizations pruning task was aborted.");
+            }
+            catch (Exception exception)
+            {
+                _logger.LogDebug(exception, "An error occurred while pruning authorizations from the database.");
```

Comment:
I would log as `Error` as this is not expected and there could be an issue that needs to be fixed.

Reply:
It's a catch all, if there are specific exceptions that you know can happen then catch it, exactly like you did above. Otherwise Error them.

Reply:
Alright, let's go with `Error` then (it shouldn't be super frequent anyway).